src/source/video_source.js

```diff
@@ -143,11 +154,15 @@ class VideoSource extends ImageSource {
         if (!this.texture) {
             this.texture = new Texture(context, this.video, gl.RGBA);
             this.texture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE);
-        } else if (!this.video.paused) {
+        } else if (!this.video.paused || this.needsUpdate) {
             this.texture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE);
             gl.texSubImage2D(gl.TEXTURE_2D, 0, 0, 0, gl.RGBA, gl.UNSIGNED_BYTE, this.video);
         }
 
+        if (this.needsUpdate) {
+            this.textureDrawn = true;
```

Comment:
What is this property for? It seems like it's only used in the check in `hasTransition`?